[CI] GeoIpCacheTests testPurgeCacheEntriesForDatabase failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #8850 / windows-2025_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/lhtmg7j274e6y)
- [elasticsearch-periodic-platform-support #8850 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/6bgzulh2phvvs)
- [elasticsearch-periodic-platform-support #8844 / windows-2025_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/4enwfzobxsrbm)
- [elasticsearch-periodic-platform-support #8844 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/p2pwkrd5lzcdm)
- [elasticsearch-periodic-platform-support #8838 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/sjerhzgmw53qy)
- [elasticsearch-periodic-platform-support #8838 / windows-2025_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/bi5myvkrmyvwg)

**Reproduction Line:**
```
gradlew ":modules:ingest-geoip:test" --tests "org.elasticsearch.ingest.geoip.GeoIpCacheTests.testPurgeCacheEntriesForDatabase" -Dtests.seed=149993FC2FC4DA5 -Dtests.locale=ff-Latn-MR -Dtests.timezone=Pacific/Port_Moresby -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.ingest.geoip.GeoIpCacheTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testPurgeCacheEntriesForDatabase),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected null, but was:<Mock for AbstractResponse, hashCode: 1755769235>
```

**Issue Reasons:**
- [main] 2 consecutive failures in test testPurgeCacheEntriesForDatabase
- [main] 3 consecutive failures in step windows-2025_checkpart1_platform-support-windows
- [main] 3 consecutive failures in step windows-2022_checkpart1_platform-support-windows
- [main] 6 failures in test testPurgeCacheEntriesForDatabase (3.1% fail rate in 192 executions)
- [main] 3 failures in step windows-2025_checkpart1_platform-support-windows (100.0% fail rate in 3 executions)
- [main] 3 failures in step windows-2022_checkpart1_platform-support-windows (100.0% fail rate in 3 executions)
- [main] 3 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpCacheTests.java
+++ b/modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpCacheTests.java
@@ -17,6 +17,7 @@ import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.ingest.geoip.stats.CacheStats;
 import org.elasticsearch.test.ESTestCase;
 
+import java.nio.file.Path;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.function.Function;
@@ -140,23 +141,24 @@ public class GeoIpCacheTests extends ESTestCase {
         GeoIpCache cache = new GeoIpCache(100);
         ProjectId projectId1 = randomUniqueProjectId();
         ProjectId projectId2 = randomUniqueProjectId();
-        String databasePath1 = "path/to/db1";
-        String databasePath2 = "path/to/db2";
+        // Turn the path strings into Paths to ensure that
+        Path databasePath1 = PathUtils.get("path/to/db1");
+        Path databasePath2 = PathUtils.get("path/to/db2");
         String ip1 = "127.0.0.1";
         String ip2 = "127.0.0.2";
 
         AbstractResponse response = mock(AbstractResponse.class);
-        cache.putIfAbsent(projectId1, ip1, databasePath1, ip -> response); // cache miss
-        cache.putIfAbsent(projectId1, ip2, databasePath1, ip -> response); // cache miss
-        cache.putIfAbsent(projectId2, ip1, databasePath1, ip -> response); // cache miss
-        cache.putIfAbsent(projectId1, ip1, databasePath2, ip -> response); // cache miss
-        cache.purgeCacheEntriesForDatabase(projectId1, PathUtils.get(databasePath1));
+        cache.putIfAbsent(projectId1, ip1, databasePath1.toString(), ip -> response); // cache miss
+        cache.putIfAbsent(projectId1, ip2, databasePath1.toString(), ip -> response); // cache miss
+        cache.putIfAbsent(projectId2, ip1, databasePath1.toString(), ip -> response); // cache miss
+        cache.putIfAbsent(projectId1, ip1, databasePath2.toString(), ip -> response); // cache miss
+        cache.purgeCacheEntriesForDatabase(projectId1, databasePath1);
         // should have purged entries for projectId1 and databasePath1...
-        assertNull(cache.get(projectId1, ip1, databasePath1));
-        assertNull(cache.get(projectId1, ip2, databasePath1));
+        assertNull(cache.get(projectId1, ip1, databasePath1.toString()));
+        assertNull(cache.get(projectId1, ip2, databasePath1.toString()));
         // ...but left the one for projectId2...
-        assertSame(response, cache.get(projectId2, ip1, databasePath1));
+        assertSame(response, cache.get(projectId2, ip1, databasePath1.toString()));
         // ...and for databasePath2:
-        assertSame(response, cache.get(projectId1, ip1, databasePath2));
+        assertSame(response, cache.get(projectId1, ip1, databasePath2.toString()));
     }
 }

--- a/modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpCacheTests.java
+++ b/modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpCacheTests.java
@@ -141,7 +141,8 @@ public class GeoIpCacheTests extends ESTestCase {
         GeoIpCache cache = new GeoIpCache(100);
         ProjectId projectId1 = randomUniqueProjectId();
         ProjectId projectId2 = randomUniqueProjectId();
-        // Turn the path strings into Paths to ensure that
+        // Turn the path strings into Paths to ensure that we always use the canonical string representation (this string literal does not
+        // round-trip when converting to a Path and back again on Windows):
         Path databasePath1 = PathUtils.get("path/to/db1");
         Path databasePath2 = PathUtils.get("path/to/db2");
         String ip1 = "127.0.0.1";

